### PR TITLE
Feature/#2 generate question

### LINKED ISSUE #2: Logic | Generating a question for each game mode
old hash:  #5
Acceptance criteria

- [x] Function generates new question for given game type (People / Starships / Vehicles)

- [x] Function will work as follows:

1. Generates 4 random and unique id in range of type repostirory (i.e. for People it is 1-82)
2. From these 4 numbers picks one random, which will be correct answer.
3. For each of numbers fetch data from SWApi, i.e. for number 50 and People type https://swapi.dev/api/people/50/
4. For correct answer fetch image using FETCH API and path (i.e. people - static/assets/img/modes/people/${id}.jpg
5. Image should be coded as string in base64 format.

 

- [x] Object should looks like follows:

```
{
"image": "base64imagestring",
"answers": ["Luke Skywalker", "Leia Organa", "Darth Vader", "Darth Sidious"],
"rightAnswer": "Darth Vader"
}
```

- [x]  Unit tests for implemented functionality.

### PR DESCRIPTION
**Pull request by Enessetere**
closes #2 
Add gathering data from API and generate quiz question based on data.

### CODE REVIEW

File: src/app/utils/quiz.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    });
+}
+
+function generateQuestion() {
+  if (questionScope.length === 0) {
+    return;
+  }
+  const ids = [];
+  while (ids.length !== 4) {
+    let id = pickRandomId();
```
Comment (paula0403) [inline src/app/utils/quiz.js:54]: Maybe You can use const instead let?

File: src/app/utils/quiz.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  await fetch(api)
+    .then((item) => item.json())
+    .then((item) => {
+      questionScope.push(...item.results);
+      if (item.next !== null) {
+        fetchItems(item.next);
+      }
+    });
+}
+
```
Comment (paula0403) [inline src/app/utils/quiz.js:16]: This empty line is unnecessary.

File: src/app/utils/quiz.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      reader.onload = () => (question.image = reader.result);
+      reader.readAsDataURL(blob);
+    });
+}
+
+function generateQuestion() {
+  if (questionScope.length === 0) {
+    return;
+  }
+  const ids = [];
```
Comment (paula0403) [inline src/app/utils/quiz.js:52]: In my opinion the name of this variable is misleading.

File: src/app/utils/quiz.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+async function getUrlData(id) {
+  await fetch(`/static/assets/img/modes/${type}/${id}.jpg`)
+    .then((res) => res.blob())
+    .then((blob) => {
+      const reader = new FileReader();
+      reader.onload = () => (question.image = reader.result);
+      reader.readAsDataURL(blob);
+    });
+}
+
```
Comment (paula0403) [inline src/app/utils/quiz.js:47]: empty line; no semicolon

**Review by paula0403 — APPROVED** [review]
-You can use const variables in loops if you don't change them,
-remember about semicolons,
-I think Your code will better without empty lines.
-good job!

File: src/app/utils/quiz.js (hunk 2 above)
Comment (DominikNowak) [reply src/app/utils/quiz.js:16]: Empty line should be in the end of every file. It is properly formatted.

File: src/app/utils/quiz.js (hunk 1 above)
Comment (DominikNowak) [reply src/app/utils/quiz.js:54]: Agree

File: src/app/utils/quiz.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  }
+  getUrlData(ids[correctIdIdx]);
+  question.answers = ids
+    .map((id) => questionScope[id])
+    .map((item) => item.name);
+  question.rightAnswer = questionScope[ids[correctIdIdx]]['name'];
+}
+
+export const initGame = (mode, url) => {
+  baseApi = url || 'https://swapi.dev/api';
```
Comment (DominikNowak) [inline src/app/utils/quiz.js:71]: You should use SW_API_BASE_URL from .env

**Review by DominikNowak — CHANGES_REQUESTED** [review]
I think that you don't have eslint configured. It's showing a few errors.

**Review by paula0403 — APPROVED** [review]
Now it's ok.

**Review by DominikNowak — APPROVED** [review]
Nicely done, good work :)

**Pull request merged**